neat_core.c

```diff
@@ -804,6 +1035,210 @@ neat_open(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port)
     return neat_he_lookup(mgr, flow, he_connected_cb);
 }
 
+neat_error_code
+neat_open_localname(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port, const char *localname)
+{
+    neat_log(NEAT_LOG_DEBUG, "%s", __func__);
+
+    if (flow->name) {
+        return NEAT_ERROR_BAD_ARGUMENT;
+    }
+
+    if (localname == NULL)
+	return neat_open(mgr, flow, name, port);
+
+    return NEAT_ERROR_UNABLE;
+}
+
+neat_error_code
+neat_open_multistream(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port, const char* localname, int count)
```

Comment:
C does not support arguments with default values to my knowledge. Optional arguments could be implemented using a variadic function (... in the parameter list and `va_list` et. al.), but I am opposed to this approach. I'd rather have multiple functions with different names instead of potentially having programs that compile just fine yet displays unusual/undesirable behavior just because optional parameters were specified incorrectly or in the wrong order. I vote for using the little static checking there is in C, at least.